File: Source/Obsequy.Model/Form/AdministratorRegistrationForm.cs
using FluentValidation.Results;
using Obsequy.Utility;

namespace Obsequy.Model
{
	public class AdministratorRegistrationForm
	{
		#region Properties

		public string Id { get; set; }

		public AdminMember Member { get; set; }

		public string Password { get; set; }
		public string ConfirmPassword { get; set; }

		#endregion

		public AdministratorRegistrationForm()
		{
			Member = new AdminMember();
		}

		public ValidationResult Validate(AccountSession accountSession, ValidationMode validationMode)
		{
			return ((new AdministratorRegistrationFormValidator(accountSession, validationMode)).Validate(this) as ValidationResult);
		}
	}
}

File: Source/Obsequy.Web/App/Forms/administratorRegistrationForm.js


angular.module('administratorRegistration-form', [])
	.factory('AdministratorRegistrationForm', ['AdminMember', function (AdminMember) {
		var AdministratorRegistrationForm = function (properties) {

			angular.extend(this, {

				member: new AdminMember(),
				password: undefined,
				confirmPassword: undefined,
				hasAcceptedEULA: undefined,

				update: function (properties) {
					properties = properties || {};

					// update properties
					this.member.update(properties.member);
					this.password = properties.password;
					this.confirmPassword = properties.confirmPassword;
					this.hasAcceptedEULA = properties.hasAcceptedEULA;
				},

				toJSON: function () {
					return {
						member: this.member.toJSON(),
						password: this.password,
						confirmPassword: this.confirmPassword,
						hasAcceptedEULA: this.hasAcceptedEULA
					};
				}
			});

			this.update(properties);
		};

		return AdministratorRegistrationForm;
	}]);